cmd/initializer/internal/config_generator.go

```diff
@@ -202,5 +205,9 @@ func (c *ConfigGenerator) allocateNodeID() (uint64, error) {
 		return 0, fmt.Errorf("invalid hostname format '%s'", name)
 	}
 
+	if c.DatanodeGroupID >= 0 {
+		nodeID = uint64(c.DatanodeGroupID)<<32 | nodeID
+	}
+
```

Comment:
Suggest add some unit test for DatanodeGroupID: https://github.com/GreptimeTeam/greptimedb-operator/blob/main/cmd/initializer/internal/config_generator_test.go